### PR TITLE
State: add getUnconnectedSiteUser selector and include it in the JPO steps.

### PR DESCRIPTION
**Pull request by AnnaMag**
This PR introduces an extra selector for the Jetpack Onboarding steps:
`getUnconnectedSiteUser()` retrieves the userEmail field of an onboarding site credentials by its ID and returns a hashed version of it.

This PR also incorporates the selector in the JPO steps.

Part of #21686.

**To test:**
- Checkout this branch
- Verify that the tests pass:

`npm run test-client client/state/selectors/test/get-unconnected-site-user-hash.js`

- also, follow the steps on https://github.com/Automattic/wp-calypso/pull/21626 to verify that the event tracking object contains all the properties set there.

### CODE REVIEW

File: client/state/selectors/get-unconnected-site-user-hash.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import createSelector from 'lib/create-selector';
+import { getUnconnectedSite } from 'state/selectors';
+
+/**
+ * Returns a hashed userEmail of a Jetpack onboarding site indexed by the siteID.
+ * Returns null if no email has been provided.
+ *
+ * @param  {Object}   state 		Global state tree.
+ * @param  {String}   siteId		SiteId of the unconnected site.
+ * @return {?String}		    		Hashed userEmail of the unconnected site.
```
Comment (ockham) [inline client/state/selectors/get-unconnected-site-user-hash.js]: Nitpick: Whitespace :slightly_smiling_face:

File: client/state/selectors/get-unconnected-site-user-hash.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+import createSelector from 'lib/create-selector';
+import { getUnconnectedSite } from 'state/selectors';
+
+/**
+ * Returns a hashed userEmail of a Jetpack onboarding site indexed by the siteID.
+ * Returns null if no email has been provided.
+ *
+ * @param  {Object}   state 		Global state tree.
+ * @param  {String}   siteId		SiteId of the unconnected site.
```
Comment (ockham) [inline client/state/selectors/get-unconnected-site-user-hash.js]: Should be `{Number}`, no?

File: client/state/selectors/test/get-unconnected-site-user-hash.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+						siteUrl: 'http://yourgroovydomain.com',
+					},
+				},
+			},
+		}, 2916284 );
+
+		expect( selected ).toBeNull();
+	} );
+
+	test( 'should return the userEmail if specified', () => {
```
Comment (ockham) [inline client/state/selectors/test/get-unconnected-site-user-hash.js]: Maybe change wording to `should return the hashed userEmail if it is specified` (or something like that).

Comment (AnnaMag) [reply client/state/selectors/test/get-unconnected-site-user-hash.js]: yes, changed after the exact same comment of Marin 😄

**Comment by AnnaMag** [comment]
Thanks @ockham and @tyxla!

**Pull request merged**